### ISSUE TITLE
max_det parameter not working in rt-detr

### ISSUE THREAD

**Issue by RomRoc** [ultralytics/ultralytics#3286]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection

### Bug

Hello, I trained a model using rt-detr.
I run inference with trained model:

`results = model('*.jpg', max_det=1000, imgsz=2048)
`
But I keep only 300 objects.
I trained same dataset with yolov8, and it detects correctly up to 1000 objects.

Can you confirm I can change object detected using this parameter?

Thanks a lot.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@RomRoc I think this is a structural limitation of RTDETR rather than a bug. @Laughing-q is that right?

**Comment by glenn-jocher** [comment]
@Laughing-q or maybe we just need to connect `max_det` arg here? https://github.com/ultralytics/ultralytics/blob/d8701b42caeb9f7f1de5fd45e7c3f3cf1724ebb6/ultralytics/nn/modules/head.py#L167

**Comment by glenn-jocher** [comment]
@RomRoc can you try changing this value manually? If that works I'll submit a PR to connect this to the `max_det` arg. Thanks!

**Comment by glenn-jocher** [comment]
@Laughing-q yeah that's what I thought. Maybe we should parameterize it so a user could start a new training with nq=1000?

**Comment by RomRoc** [comment]
Thanks a lot for your quick answers, let me know if I can help you somehow.
Do you plan to modify code to allow more than 300 objects?
